packages/@react-spectrum/s2/src/TreeView.tsx

```diff
@@ -0,0 +1,500 @@
+  hasChildItems?: boolean,
+  /** A list of child tree item objects used when dynamically rendering the tree item children. */
+  childItems?: Iterable<T>
+}
+
+interface TreeRendererContextValue {
+  renderer?: (item) => ReactElement<any, string | JSXElementConstructor<any>>
+}
+const TreeRendererContext = createContext<TreeRendererContextValue>({});
+
+function useTreeRendererContext(): TreeRendererContextValue {
+  return useContext(TreeRendererContext)!;
+}
+
+
+let InternalTreeContext = createContext<{isDetached?: boolean, isEmphasized?: boolean}>({});
+
+// TODO: the below is needed so the borders of the top and bottom row isn't cut off if the TreeView is wrapped within a container by always reserving the 2px needed for the
+// keyboard focus ring. Perhaps find a different way of rendering the outlines since the top of the item doesn't
+// scroll into view due to how the ring is offset. Alternatively, have the tree render the top/bottom outline like it does in Listview
+const tree = style({
+  userSelect: 'none',
+  minHeight: 0,
+  minWidth: 0,
+  width: 'full',
+  overflow: 'auto',
+  boxSizing: 'border-box',
+  justifyContent: {
+    isEmpty: 'center'
+  },
+  alignItems: {
+    isEmpty: 'center'
+  },
+  height: {
+    isEmpty: 'full'
+  },
+  '--indent': {
+    type: 'width',
+    value: 16
+  }
+}, getAllowedOverrides({height: true}));
+
+function TreeView(props: TreeViewProps, ref: DOMRef<HTMLDivElement>) {
+  let {children, isDetached, isEmphasized} = props;
+  isEmphasized = !isDetached;
+
+  let renderer;
+  if (typeof children === 'function') {
+    renderer = children;
+  }
+
+  let domRef = useDOMRef(ref);
+
+  let layout = useMemo(() => {
+    return new UNSTABLE_ListLayout({
+      rowHeight: isDetached ? 42 : 40
+    });
+  }, [isDetached]);
+
+  return (
+    <UNSTABLE_Virtualizer layout={layout}>
+      <TreeRendererContext.Provider value={{renderer}}>
+        <InternalTreeContext.Provider value={{isDetached, isEmphasized}}>
+          <UNSTABLE_Tree
+            {...props}
+            className={({isEmpty}) => tree({isEmpty, isDetached}, props.styles)}
+            selectionBehavior="toggle"
+            ref={domRef}>
+            {props.children}
+          </UNSTABLE_Tree>
+        </InternalTreeContext.Provider>
+      </TreeRendererContext.Provider>
+    </UNSTABLE_Virtualizer>
+  );
+}
+
+const selectedBackground = lightDark(colorMix('gray-25', 'informative-900', 10), colorMix('gray-25', 'informative-700', 10));
+const selectedActiveBackground = lightDark(colorMix('gray-25', 'informative-900', 15), colorMix('gray-25', 'informative-700', 15));
+
+const rowBackgroundColor = {
+  default: '--s2-container-bg',
+  isFocusVisibleWithin: colorMix('gray-25', 'gray-900', 7),
+  isHovered: colorMix('gray-25', 'gray-900', 7),
+  isPressed: colorMix('gray-25', 'gray-900', 10),
+  isSelected: {
+    default: colorMix('gray-25', 'gray-900', 7),
+    isEmphasized: selectedBackground,
```

Comment:
Does this do anything? I could not tell the difference between emphasized and non-emphasized in the storybook. Looks like maybe this is somehow attached to detached instead?

Reply:
correct, isEmphasized is === !isDetached permanently right now until designs work for both. I can remove the prop for `isEmphasized` if that's more helpful

Reply:
looks like Figma has both. what am I missing?

Reply:
i've removed the restriction, it doesn't seem as bad as I remembered in terma of contrast